src/util/server.c

```diff
@@ -319,7 +315,7 @@ static void setup_signals(void)
     BlockSignals(false, SIGHUP);
     BlockSignals(false, SIGTERM);
 
-#ifndef HAVE_PRCTL
+#ifdef HAVE_NO_PRCTL_WHATSOEVER
```

Comment:
I have no idea why this was conditioned by `ifndef HAVE_PRCTL` (and git history doesn't help me much).

@sumit-bose, do you have an idea?

Reply:
Hi,

I guess it was just best effort trying and hoping that SIGSEGV or SIGABRT will be send if the parent dies and `prctl()` is not available. If FreeBSD has a better solution this block can still be used for other platforms as a fallback.

bye,
Sumit

Reply:
By matter of fact recent versions of SSSD require `prctl()` unconditionally (and this PR will add support of `procctl()`)

I think we should just clearly hard-require it (any option) at ./configure step, and remove this fallback altogether.